source/core/slang-zip-file-system.h

```diff
@@ -0,0 +1,35 @@
+#ifndef SLANG_ZIP_FILE_SYSTEM_H
+#define SLANG_ZIP_FILE_SYSTEM_H
+
+#include "slang-basic.h"
+
+#include "../../slang-com-ptr.h"
+
+namespace Slang
+{
+
+class CompressedFileSystem : public RefObject, public ISlangMutableFileSystem
+{
+public:
+
+    enum class CompressionType
+    {
+        BestSpeed,
+        BestCompression,
+    };
+
+        /// Get as an archive (that can be saved to disk)
+    virtual ArrayView<uint8_t> getArchive() = 0;
```

Comment:
Should this be defined to return a "blob" that snapshots the current contents of the archive?

I was about to say that such a thing would be easy with our `Array<T>` type, but then I realized that I only changed our `String` to be copy-on-write, and not `Array<T>`.

With `String` it should be relatively easy at this point to change the heap-allocated `StringRepresentation` to support the `IBlob` interface, which would mean that creating a blob that snapshots the current state of a `String` would be a simple constant-time operation (just retain a reference to the `StringRepresnetation` and cast it to the blob interface). There'd be no extra heap allocation or copies, but then if somebody subsequently tries to mutate the original string our copy-on-write logic would automatically make a copy at that point.

If we changed `Array<T>` to use a similar implementation, where the actual `Array<T>` is a single pointer in size, and points to a heap-allocated `ArrayRepresentation` object, then we could make the operation of creating a blob from an array just as simple, and it would have the same property of being able to snapshot an immutable blob from a mutable array.

None of this is very relevant to the current PR, but it does seem a bit unsafe to be directly returning a view into the bits that make up the archive, when those bits might change with subsequent filesystem calls (or even the address of the allocation might change, right?).